src/staticgraph.jl

```diff
@@ -10,46 +10,51 @@ struct StaticGraph{T<:Integer, U<:Integer} <: AbstractStaticGraph{T, U}
     f_ind::Vector{U}
 end
 
-
 # sorted src, dst vectors
 # note: this requires reverse edges included in the sorted vector.  
-function StaticGraph(n_v, ss::AbstractVector, ds::AbstractVector)
+function StaticGraph(nv::I, ss::AbstractVector, ds::AbstractVector) where {I<:Integer}
     length(ss) != length(ds) && error("source and destination vectors must be equal length")
-    (n_v == 0 || length(ss) == 0) && return StaticGraph(UInt8[], UInt8[1])
-    f_ind = [searchsortedfirst(ss, x) for x in 1:n_v]
+    (nv == 0 || length(ss) == 0) && return StaticGraph(UInt8[], UInt8[1])
+    f_ind = [searchsortedfirst(ss, x) for x in 1:nv]
     push!(f_ind, length(ss)+1)
     T = mintype(ds)
     U = mintype(f_ind)
     return StaticGraph{T, U}(convert(Vector{T},ds), convert(Vector{U}, f_ind))
 end
 
 # sorted src, dst tuples
-function StaticGraph(n_v, sd::Vector{Tuple{T, T}}) where T <: Integer
+function StaticGraph(nv::I, sd::Vector{Tuple{T, T}}) where {T<:Integer, I<:Integer}
     ss = [x[1] for x in sd]
     ds = [x[2] for x in sd]
-    StaticGraph(n_v, ss, ds)
+    return StaticGraph(nv, ss, ds)
 end
 
 function StaticGraph(g::LightGraphs.SimpleGraphs.SimpleGraph)
     sd1 = [Tuple(e) for e in edges(g)]
     ds1 = [Tuple(reverse(e)) for e in edges(g)]
-
     sd = sort(vcat(sd1, ds1))
-    StaticGraph(nv(g), sd)
+    return StaticGraph(nv(g), sd)
+end
+
+function StaticGraph()
+    return StaticGraph(UInt8[], UInt8[1])
 end
 
 badj(g::StaticGraph, s) = fadj(g, s)
 
 ne(g::StaticGraph{T, U}) where T where U = U(length(g.f_vec) ÷ 2)
 
-function has_edge(g::StaticGraph, e::StaticGraphEdge)
+function in(e::StaticGraphEdge, g::StaticGraph)
```

Comment:
We don't define set membership for graphs - `has_edge` is the formal API contract.

Reply:
Right sorry, but there should be a `in(e::StaticGraphEdge, edges(g))` ?

Reply:
There's `in(e, es::SimpleEdgeIter) = has_edge(es.g, e)` for SimpleEdges, but those are defined in the edgeiter code, not in the main graph file.